src/lib.rs

```diff
@@ -1011,13 +1037,19 @@ impl<TRepoTypes: RepoTypes> Future for IpfsFuture<TRepoTypes> {
                             .collect();
                         let _ = ret.send(peers);
                     }
-                    IpfsEvent::FindPeer(peer_id, ret) => {
-                        let known_addrs = self.swarm.swarm.addresses_of_peer(&peer_id);
-                        let addrs = if !known_addrs.is_empty() {
-                            known_addrs
+                    IpfsEvent::FindPeer(peer_id, local_only, ret) => {
+                        let swarm_addrs = self.swarm.swarm.addresses_of_peer(&peer_id);
+                        let locally_known_addrs = if !swarm_addrs.is_empty() {
+                            swarm_addrs
                         } else {
                             self.swarm.kademlia().addresses_of_peer(&peer_id)
                         };
+                        let addrs = if !locally_known_addrs.is_empty() || local_only {
+                            Either::Left(locally_known_addrs)
+                        } else {
+                            let own_id = self.own_id.clone();
+                            Either::Right(self.swarm.get_closest_peers(own_id))
```

Comment:
surprised this is not `peer_id` but `own_id`; could this have some comment explaining why?

Reply:
not much reason other than me having a brainfart moment, I'm afraid :laughing:; fixed in the force push